doc/source/cluster/kubernetes/examples/gpu-training-example.md

```diff
@@ -0,0 +1,149 @@
+(kuberay-gpu-training-example)=
+
+# ML training with GPU on Kubernetes
+In this guide, we show you how to run a sample Ray machine learning training workload with GPU on Kubernetes infrastructure. We will run Ray's {ref}`PyTorch image training benchmark <pytorch_gpu_training_benchmark>` with a 1 gigabyte training set. The following script consists of:
+- Step 1: Setup a Kubernetes cluster on GCP.
+- Step 2: Deploy a Ray cluster on Kubernetes with the KubeRay operator.
+- Step 3: Run the PyTorch image training benchmark. 
+
+```shell
+# Step 1: Setup a Kubernetes cluster on GCP
+# Create a node-pool for a CPU-only head node
+# e2-standard-8 => 8 vCPU; 32 GB RAM
+gcloud container clusters create gpu-cluster-1 \
+    --num-nodes=1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+    --zone=us-central1-c --machine-type e2-standard-8
+
+# Create a node-pool for GPU. The node is for a GPU Ray worker node.
+# n1-standard-8 => 8 vCPU; 30 GB RAM
+gcloud container node-pools create gpu-node-pool \
+  --accelerator type=nvidia-tesla-t4,count=1 \
+  --zone us-central1-c --cluster gpu-cluster-1 \
+  --num-nodes 1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+  --machine-type n1-standard-8
+
+# Install NVIDIA GPU device driver
+kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+
+# Step 2: Deploy a Ray cluster on Kubernetes with the KubeRay operator.
+# Please make sure whether you connect to your Kubernetes cluster on GCP or not.
+
+# Create the KubeRay operator
+kubectl create -k "github.com/ray-project/kuberay/ray-operator/config/default?ref=v0.3.0&timeout=90s"
+
+# Create a Ray cluster
+kubectl apply -f https://raw.githubusercontent.com/ray-project/ray/master/doc/source/cluster/kubernetes/configs/ray-cluster.gpu.yaml
+
+# Set up port-forwarding
+kubectl port-forward services/raycluster-head-svc 8265:8265
+
+# Test the cluster
+ray job submit --address http://localhost:8265 -- python -c "import ray; ray.init(); print(ray.cluster_resources())"
+
+# Step 3: Run the PyTorch image training benchmark.
+pip3 install -U "ray[default]"
+
+# Download the Python script
+curl https://raw.githubusercontent.com/ray-project/ray/master/doc/source/cluster/doc_code/pytorch_training_e2e_submit.py -o pytorch_training_e2e_submit.py
+
+# Submit the training job to your ray cluster
+python3 pytorch_training_e2e_submit.py
+
+# Use the following command to follow this Job's logs:
+# Substitute the Ray Job's submission id.
+ray job logs 'raysubmit_xxxxxxxxxxxxxxxx' --follow
+```
+
+# Step 1: Setup a Kubernetes cluster on GCP.
```

Comment:
Sections should probably start with two hash marks `##` so that the section headings render smaller than the page's title.

Reply:
Updated.